GiniVision/Classes/Core/Screens/Multipage Review/MultipageReviewCollectionsCellPresenter.swift

```diff
@@ -0,0 +1,209 @@
+//
+//  MultipageReviewCollectionCellPresenter.swift
+//  GiniVision
+//
+//  Created by Enrique del Pozo Gómez on 12/28/18.
+//
+
+import Foundation
+
+protocol MultipageReviewCollectionCellPresenterDelegate: class {
+    func multipage(_ reviewCollectionCellPresenter: MultipageReviewCollectionCellPresenter,
+                   didUpdate cell: MultipageReviewCollectionCellPresenter.MultipageCollectionCellType,
+                   at indexPath: IndexPath)
+}
+
+final class MultipageReviewCollectionCellPresenter {
+    
+    weak var delegate: MultipageReviewCollectionCellPresenterDelegate?
+    var thumbnails: [String: [ThumbnailType: UIImage]] = [:]
+    fileprivate let giniConfiguration: GiniConfiguration
+    fileprivate let thumbnailsQueue = DispatchQueue(label: "Thumbnails queue")
+    
+    enum MultipageCollectionCellType {
+        case main(MultipageReviewMainCollectionCell, (NoticeActionType) -> Void)
+        case pages(MultipageReviewPagesCollectionCell)
+    }
+    
+    enum ThumbnailType {
+        case big, small
```

Comment:
Here, since `big` is actually 100% scale (the full image), I'd call it original. Otherwise, I'd still expect it to be downscaled at least a little.
And since the big one is the original, it's not even a thumbnail, so I'd actually call them "original" and "thumbnail" and the enum "ImageScale" or something.

Reply:
It's not the original version in many cases (mostly on iPhone). It's the size that matched the screen width.